[11.0][MIG] delivery_carrier_label_postlogistics

grindtildeath opened the pull request:
Based on #128 
Depends on : 
 - [x] #156 
 - [x] https://github.com/OCA/server-tools/pull/1261

yvaucher committed:
[IMP] rename module delivery_carrier_label_laposte => delivery_carrier_label_postlogistics

[FIX] contributor

[FIX] exception propagation

[IMP] exceptions syntax

[FIX] rename xml id to fit to new module name delivery_carrier_label_postlogistics

[FIX] invert logic of required on postlogistics_service_group_id field in carrier view

[ADD] logging info as updating service can take some times

[REF] make it easier to inherit from _generate_postlogistics_label (renamed from _generate_poste_ch_label)

[ADD] default options for label layout, output format and resolution

[ADD] file_type on shipping label

[FIX] set file format to lower case

[FIX] get code instead of option browse record in format, layout and resolution options

[ADD] fr.po

jgrandguillaume committed:
[FIX] When using update_postlogistics_options action, we must take the company contain in the current configuration (res.config) instead of the company of the user

yvaucher committed:
[IMP] adapt delivery_carrier_label_postlogistics to generate a list of label per picking tracking

[IMP] adapt to signature change in base_delivery_carrier_label for generate_pack_labels -> generate_shipping_labels

[IMP] labels on stock.picking instead of stock.picking.out so we can generate pretty PostLogistics labels even for the not yet happy customers who have to return product.

[ADD] pot file

Commit:
[ADD] missing access rights

[FIX] wrong method name

yvaucher committed:
[IMP] Permits to use multiple basic services

[IMP] postlogistics - licenses management

[FIX] wrong field name for license number

yvaucher committed:
[ADD] security groups for postlogistics licenses

[IMP] PEP8

bealdav committed:
[IMP] move postlogistic menu in submenu

Commit:
[ADD] use the new parameter 'tracking_ids' to generate only the labels of the trackings given in tracking_ids if used.

The implementation by default still generate them all at once from a button, but for instance the module allowing to print labels of a dispatch use it

use the name of the picking when there is no pack

yvaucher committed:
Consider returned sPDF file as PDF file. Those file are PDF without integrated font

[FIX] make sure pack number (which can be picking name) has no unallowed character for item id of postlogistics label

delivery_carrier_label_postlogistics - Improve carrier type selection inheritance

Update API version and when recipient partner's field street2 is defined we rewrite name and steet label address lines to print it on label.

Use AddressSuffix instead of reforging the address lines that way we don't break company name addition

Update PostLogistics webservice doc version

[FIX] fr.po file by removing ^M breaklines

[FIX] postlogistics warnings unwrapping

Improve file_type selection field inheritance

add extension to file name

Commit:
wrong docstring, trackings expects a list of browse record

yvaucher committed:
revert rev 35 for conflict resolution

yvaucher committed:
revert rev 35 for conflict resolution

Replay of Improve file_type selection field inheritance

Commit:
Launchpad automatic translations update.

yvaucher committed:
set installable to false

Commit:
Move delivery_carrier_label_postlogistics from __unported__ to root folder

Set delivery_carrier_label_postlogistics installable

Extract the description in README.rst

New API for res.company

Migration of delivery.py to the new API

The state field has been replaced by boolean fields

Migration of stock picking to the new API

And API changes in base_delivery_carrier_label

compute method must assign a recordset, not a list

Wrong assignation

Migration of res.config to new API

Fix a comment

Comma at end of line

Migrated base_delivery_carrier_label so adapt the call

Migrated here: https://github.com/bealdav/carrier-delivery/pull/1

The field on the package is now parcel_tracking

The relation is now package_id

The domain prevents other modules to show their options

The drawback of this change is that postlogistics will also
show the options of the others carriers but I can't find a way
to filter them out. The user still see in the search that an option
is not related to postlogistics.

Fallback on 'en' when no lang is provided

Update the carrier inside the loop

Get the allowed options on the parent instead of the related

The related allowed_option_ids is computed before carrier_id is set and is
incorrect. Also, a change needs to be saved to see the change back on the
related field.

Hide the option of other carriers such as colissimo

The domain in the view is insufficient because it displays the options with an
empty partner.  Adding dynamically the domain when carrier's partner is
postlogistics ensures that only postlogistics' options are shown.

Fix: W503 line break before binary operator

Readme more similar than the OCA's template

Add license key and OCA logo

Remove dead code

Implement Postlogistics Cash on Delivery labels

The webservice awaits the cash on delivery amount.
When a whole sales order is delivered in one picking / one package, we the
amount is the total of the sales order. If it is split in several pickings /
packages, we don't know the value of each package and so we ask the user to
fill the price on the packages.

Update translations

Fix error when user has no language

Set default language to 'en_US' when the user has no language

tafaRU committed:
Protect suds import with a try-except ImportError so that the presence of this module in the addons path will not cause an error even if the addon is not installed.

yvaucher committed:
Fixup 34415 to add import of HttpAthenticated in try except

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[FIX] remove duplicate key in manifest

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

yvaucher committed:
[PORT] delivery_carrier_label_postlogistics port to 9.0
- Reactivate
- Move files in views and models
- Adapt XML views
- Update README.rst

[PORT] delivery_carrier_label_postlogistics - move license config in a separate view list. This fix an issue of write on related One2many in res.config.settings object

Update README.rts with links and requirements

Add missing set default on licenses

Use short license header

Set module category

<openerp> replaced by <odoo> tag in xml files

Update README.rst

Fix postlogistics tests

Commit:
Fix assignation on recordset with len > 1

Commit:
Rename field type to carrier_type in delivery.carrier

Missing acl for stock user/manager

pedrobaeza committed:
[MIG] Make modules uninstallable

[MIG] Rename manifest files

yvaucher committed:
Use configuration_helper to simplify the settings class (#84)

Add test mode using integration server of Swiss post + however, it needs https://fedorahosted.org/suds/attachment/ticket/239/suds_recursion.patch on suds library

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

Add tracking number format option.

By default let postlogistics generate the tracking number.

The new option is to use picking number or concatenation of picking
number and pack counter.

Conflicts:
	delivery_carrier_label_postlogistics/company.py
	delivery_carrier_label_postlogistics/postlogistics/web_service.py
	delivery_carrier_label_postlogistics/res_config.py
	delivery_carrier_label_postlogistics/res_config_view.xml

Implement delivery instructions ZAW

Improve error handling for 3 default parameters layout, output format and resolution

Refactor generation of tracking number to make it overridable

Don't send unset data for delivery instruction in order to get the error message from webservice

Add a note on patch to apply + a copy of it in the module

Add nocache with suds to force the reload of the definition otherwise it
work once then fail on next request.

Concatenate all pack tracking numbers on picking carrier tracking ref field when using packs

Add option to print ProClima logo

Fix assignation of tracking ref

Fix errors on check of parameters

yvaucher committed:
[MIG] delivery_carrier_label_postlogistics to 10.0

* Simplify signature of soap connector by removing company
* As carrier inherits from product, add a service postlogistics product
* Use an image instead of base64 for postlogistics logo

yvaucher committed:
Fix missing decorator on fields_view_get

Fix tests with change of xmlid

grindtildeath committed:
Fix domain on delivery.carrier.option tmpl_option_id_domain field. Use delivery_type instead of carrier_type.

Use partner_id instead of product to create template options

Define postlogistics_ methods

Add to contributors

Fix travis

Improvements after review

Define new readme folder for generator

Fix manifest, import, rst, etc

Automatically patch suds library on module loading

This allows to use suds library without having to patch manually for recursion issue.
Moreover it logs any HTTP error and displays it with a meaningful error.

grindtildeath committed:
[MIG] delivery_carrier_label_postlogistics : Migration to v11

arkostyuk commented on delivery_carrier_label_postlogistics/models/delivery.py:
```diff
@@ -109,18 +109,20 @@ class DeliveryCarrierOption(models.Model):
-            domain = self.env.context.get('default_tmpl_option_id_domain')
-            option.tmpl_option_id_domain = (
-                option.carrier_id.allowed_options_domain or domain or '[]')
+            domain = self.env.context.get('default_allowed_tmpl_options_ids')
```
doesn't appear to me that `domain` is the correct name for the content of it now

arkostyuk commented on delivery_carrier_label_postlogistics/models/delivery.py:
```diff
@@ -109,18 +109,20 @@ class DeliveryCarrierOption(models.Model):
-                option.carrier_id.allowed_options_domain or domain or '[]')
+            domain = self.env.context.get('default_allowed_tmpl_options_ids')
+            option.allowed_tmpl_options_ids = (
+                option.carrier_id.allowed_tmpl_options_ids or domain or '[]')
```
`or '[]'` seems quite dangerous to use here, since you're writing to `Many2many` :smile:

tschanzt reviewed (commented):
There is a lot of commented xml around. I think you should add a comment why it is still needed or it should be dropped.

grindtildeath committed:
Remove old wsdl and xsd files, change manual to english last version

asaunier commented:
Perhaps you could squash the old consecutive commits of the same authors?

grindtildeath commented:
@asaunier I will do it as soon as https://github.com/OCA/delivery-carrier/pull/128 is merged, but I prefer keeping it so if more is to be changed.

grindtildeath committed:
Add default values for invoice_policy and expense_policy

grindtildeath committed:
Fix flake8 / pylint

simahawk commented:
https://github.com/OCA/server-tools/pull/1261 merged

Build red

sebalix committed:
[IMP] delivery_carrier_label_postlogistics: improve test coverage

sebalix committed:
[FIX] delivery_carrier_label_postlogistics: display Postlogistics fields correctly on the delivery carrier form

grindtildeath committed:
Improve tests with application of options on picking

grindtildeath commented:
@simahawk @yvaucher @tschanzt @sebalix @asaunier 
Can you guys please pass byy for a review and get this old PR merged ?

sebalix commented on oca_dependencies.txt:
```diff
@@ -0,0 +1,7 @@
+server-tools
+stock-logistics-workflow
+webkit-tools
+web
```
We only need `server-tools` here (the module depends on `configuration_helper` from OCA/server-tools and `base_delivery_carrier_label` depending on `delivery` only which is a standard one).

pedrobaeza replied:
Please attend this comment for not making slower this CI

grindtildeath committed:
[IMP] Consider parent partner name if partner name is empty

When using a specific partner contact (e.g. shipping address) as recipient,
the name of the partner can be empty. In such case, the name of its parent
must be used.

sebalix commented:
@grindtildeath can you cherry-pick the commit from https://github.com/OCA/delivery-carrier/pull/191 too?

yvaucher commented:
/ocabot merge

grindtildeath committed:
Fix and generate README.rst

yvaucher commented:
/ocabot merge nobump
